h2/src/main/org/h2/mvstore/MVMap.java

```diff
@@ -13,62 +13,85 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicReference;
 import org.h2.mvstore.type.DataType;
 import org.h2.mvstore.type.ObjectDataType;
+import org.h2.mvstore.type.StringDataType;
 
 /**
  * A stored map.
  * <p>
- * Read operations can happen concurrently with all other
+ * All read and write operations can happen concurrently with all other
  * operations, without risk of corruption.
- * <p>
- * Write operations first read the relevant area from disk to memory
- * concurrently, and only then modify the data. The in-memory part of write
- * operations is synchronized. For scalable concurrent in-memory write
- * operations, the map should be split into multiple smaller sub-maps that are
- * then synchronized independently.
  *
  * @param <K> the key class
  * @param <V> the value class
  */
 public class MVMap<K, V> extends AbstractMap<K, V>
-        implements ConcurrentMap<K, V> {
+        implements ConcurrentMap<K, V>, Cloneable {
```

Comment:
Yes, I dropped that idea in favor of cloneIt(), but failed to remove interface. This is needed because we need a generic copy constructor for R/O copy of the map in openReadOnly()